mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java

```diff
@@ -124,14 +131,26 @@
         return ResponseEntity.noContent().build();
     }
 
+    /**
+     * Checks if a service is deployed and throws a ResponseStatusException with the http status CONFLICT (409) if
+     * the service is deployed.
+     * @param service Checks if this service is deployed
+     * @throws KubernetesResourceException if the service is deployed. It uses the http status CONFLICT
+     */
+    private void throwConflictIfServiceIsDeployed(MicoService service) throws KubernetesResourceException {
+        if (micoKubernetesClient.isServiceDeployed(service)) {
+            log.info("Micoservice '{}' in version '{}' is deployed. It is not possible to delete a deployed service.", service.getShortName(), service.getVersion());
+            throw new ResponseStatusException(HttpStatus.CONFLICT, "Service is currently deployed!");
+        }
+    }
+
     @DeleteMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}")
-    public ResponseEntity<Void> deleteAllVersionsOfService(@PathVariable(PATH_VARIABLE_SHORT_NAME) String shortName) {
+    public ResponseEntity<Void> deleteAllVersionsOfService(@PathVariable(PATH_VARIABLE_SHORT_NAME) String shortName) throws KubernetesResourceException {
         List<MicoService> micoServiceList = getAllVersionsOfServiceFromDatabase(shortName);
         log.debug("Got following services from database: {}", micoServiceList);
-        micoServiceList.forEach(service -> {
-            //TODO: check for deployment
-        });
-
+        for(MicoService micoService : micoServiceList){
+            throwConflictIfServiceIsDeployed(micoService);
```

Comment:
Couldn't we have done this function call inside the lambda forEach functionality as well?

Reply:
I think this is not possible because of the way lambda expressions and checked exceptions work in java.